File: backend/app/schemas/change.py
from pydantic import BaseModel
from datetime import date, datetime


class ChangeCreate(BaseModel):
    title: str
    description: str
    change_type: str  # scope, time, cost, resource
    impact: str | None = None
    requested_by: str | None = None
    request_date: date


class ChangeUpdate(BaseModel):
    title: str | None = None
    description: str | None = None
    change_type: str | None = None
    impact: str | None = None
    status: str | None = None
    comments: str | None = None


class ChangeResponse(BaseModel):
    id: int
    folio: str
    title: str
    description: str
    change_type: str
    impact: str | None
    requested_by: str | None
    request_date: date
    status: str
    approved_by_id: int | None
    approval_date: date | None
    comments: str | None
    project_id: int
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/change.ts
/**
 * Change request type definitions.
 * Aligned with backend schemas: ChangeResponse.
 */

/** Matches backend ChangeResponse schema */
export interface Change {
  id: number;
  folio: string;
  title: string;
  description: string | null;
  change_type: string;
  impact: string | null;
  requested_by: string | null;
  request_date: string | null;
  status: string;
  approved_by_id: number | null;
  approval_date: string | null;
  comments: string | null;
  project_id: number;
  created_at: string;
}

/** Cross-project change view used in ChangesPage (with project info) */
export interface ChangeWithProject {
  id: number;
  folio: string;
  title: string;
  change_type: string;
  impact: string;
  requested_by: string;
  status: string;
  project_name: string;
  project_id: number;
  request_date: string;
}
